### DISCUSSION TITLE
should tycho copy over the pgp keys by default? (or should i somewhere configure something?)

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-tycho/tycho#2855] [General]
Hi,

see the discussion here: https://github.com/eclipse-equinox/p2/discussions/333

i just wonder how i can fix this for our product, it seems tycho can do this? copy over the pgp keys to our p2 repo that tycho generates (where yes are all kinds of eclipse shipped plugins are also in)

**Comment by laeubi** [comment]
I think currently there is at least no test for such feature, so the best is to provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate this and we can see if it already works. If not it seems useful to add this (in general Tycho *should* copy properties if things are mirrored).

**Reply by jcompagner** [reply]
as i replied here: https://github.com/eclipse-equinox/p2/discussions/333#discussioncomment-7103045
i am still not sure where the problem really is, because it seems that the source (2022.03 install) is the problem not the target (2023.09) where i want to go to..

**Reply by jcompagner** [reply]
how should that look like?
https://download.servoy.com/developer/latest/releasecandidate/artifacts.jar

if i look in there the some jars have a "pgp.signatures" and a "pgp.publicKeys" (like org.eclipse.jetty.io or many of the org.osgi group)
but by far not all eclipse jars for example all "org.eclipse.swt.*" plugins don't have any of that.

**Reply by HannesWell** [reply]
> but by far not all eclipse jars for example all \"org.eclipse.swt.*\" plugins don't have any of that.

That's because only third-party dependencies are pgp-signed (IIRC only if they are not jar-signed) while the bundles produced by Eclipse-Platform/Equinox/JDT/PDE are only jar-signed by Eclipse.
